### PR TITLE
Author scrolling ui

### PR DESCRIPTION
**Pull request by BaileyMcKelway**
Here's a prototype of the scrolling feature. 

How it is working:
- Clicking on the button by user name will scroll to the paragraph that the user has edited.
- On the first scroll the edit will scroll to the top of the Article Viewer
- Afterwards, the next edit that is out of view of the Article Viewer will be scrolled to the bottom position of the Article Viewer

-If the user switches scrolling from one student to another then the scroll feature will scroll to the new students edit that is below the current edit
- If there are no edits below the current edit then it scrolls to the first edit by the new student

Basically, the functionality is like "find on page"

### CODE REVIEW

**Comment by BaileyMcKelway** [comment]
Apparently, I can't upload videos to GitHub but here is a link to a demo

[https://youtu.be/DTxxZsR0m74](url)

**Comment by ragesoss** [comment]
Whoops! Looks like I never submitted my quick initial review from last week. The only remaining relevant part is:

Can these new functions related to the scrolling feature be extracted to their own file? They've got a lot going on internally, but most of it is not directly the concern of the legend, so this will be a lot easier to make sense of the code for author scrolling is isolated from the other things that the legend does.

**Comment by BaileyMcKelway** [comment]
Will do! Should I just create a class inside of the utils folder? "app/assets/javascripts/components/common/ArticleViewer/utils"

 I also want to run some more tests on it.

**Comment by ragesoss** [comment]
Yes, that sounds like a good place for it.

re: tests, great!

**Comment by ragesoss** [comment]
This works really nicely! I'm going to merge and deploy it now. Thanks Bailey, this is a really sweet feature!

**Comment by ragesoss** [comment]
@BaileyMcKelway It's live now, eg, https://dashboard.wikiedu.org/courses/Georgetown_University/Culture_Medicine_and_Gender_(Spring_2020)/students/articles/Ddanaie2021?showArticle=51302794

One little bug: upon receiving the authorship highlighting information, it briefly flashes some additional text in the legend (maybe the message about no revisions?) before rendering the scroll buttons.
